api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/components/ParameterSubstitutor.java

```diff
@@ -0,0 +1,52 @@
+package edu.cornell.mannlib.vitro.webapp.dynapi.components;
+
+import java.util.Set;
+
+import edu.cornell.mannlib.vitro.webapp.dynapi.components.ParameterSubstitution.Direction;
+import edu.cornell.mannlib.vitro.webapp.dynapi.data.Data;
+import edu.cornell.mannlib.vitro.webapp.dynapi.data.DataStore;
+
+public class ParameterSubstitutor {
+
+    public static void forwardSubstitution(Set<ParameterSubstitution> substitutions, DataStore dataStore) {
+        for (ParameterSubstitution substitution : substitutions) {
+            if (substitution.containsDirection(Direction.FORWARD)) {
+                String sourceName = substitution.getSource().getName();
+                if (dataStore.contains(sourceName)) {
+                    substitute(substitution.getSource(), substitution.getTarget(), dataStore);
+                }
+            }
+        }
+    }
+
+    public static void backwardSubstitution(Set<ParameterSubstitution> substitutions, DataStore dataStore) {
+        for (ParameterSubstitution substitution : substitutions) {
+            if (substitution.containsDirection(Direction.BACKWARD)) {
+                String targetName = substitution.getTarget().getName();
+                if (dataStore.contains(targetName)) {
+                    substitute(substitution.getTarget(), substitution.getSource(), dataStore);
+                }
+            }
+        }
+    }
+
+    private static void substitute(Parameter source, Parameter target, DataStore dataStore) {
+        Data data = dataStore.getData(source.getName());
+        data.setParam(target);
+        dataStore.remove(source.getName());
+        dataStore.addData(target.getName(), data);
```

Comment:
what if target.getName() exists in data store for other steps? it will be overridden? can we make this substitution only local for an operationalStep, not impacting complete dataStore?